In [1]:
import numpy as np
import random
import time

Проанализируем выигрышные комбинации:

777: 200 руб.\
999: 100 руб.\
555: 50 руб.\
333: 15 руб.\
111: 10 руб.\
77: 5 руб.\
7: 3 руб.\
00: 2 руб.\
0: 1 руб.

In [16]:
win_combination = 0
for x in range(0, 1000):
    x = '{:03}'.format(x)
    win_numbers = ['999', '777', '555', '333', '111']   
    if x in win_numbers or x.endswith('0') or x.endswith('00') or x.endswith('7') or x.endswith('77'):
        win_combination += 1
        print(x, end = " ")

print(' ') 
print('Количество выигрышных комбинаций', win_combination)

000 007 010 017 020 027 030 037 040 047 050 057 060 067 070 077 080 087 090 097 100 107 110 111 117 120 127 130 137 140 147 150 157 160 167 170 177 180 187 190 197 200 207 210 217 220 227 230 237 240 247 250 257 260 267 270 277 280 287 290 297 300 307 310 317 320 327 330 333 337 340 347 350 357 360 367 370 377 380 387 390 397 400 407 410 417 420 427 430 437 440 447 450 457 460 467 470 477 480 487 490 497 500 507 510 517 520 527 530 537 540 547 550 555 557 560 567 570 577 580 587 590 597 600 607 610 617 620 627 630 637 640 647 650 657 660 667 670 677 680 687 690 697 700 707 710 717 720 727 730 737 740 747 750 757 760 767 770 777 780 787 790 797 800 807 810 817 820 827 830 837 840 847 850 857 860 867 870 877 880 887 890 897 900 907 910 917 920 927 930 937 940 947 950 957 960 967 970 977 980 987 990 997 999  
Количество выигрышных комбинаций 204


Рассчитаем вероятность выигрыша

In [21]:
probability = win_combination/1000


print('Вероятность выигрыша составляет', probability)

Вероятность выигрыша составляет 0.204


Методом Монте Карло рассчитаем средний выигрыш

In [28]:
prize = 0
prize_0 = 0
prize_00 = 0
prize_7 = 0
prize_77 = 0
prize_111 = 0
prize_333 = 0
prize_555 = 0
prize_777 = 0
prize_999 = 0
winning_combination = 204

for i in range(0, 1000):
    i = '{:03}'.format(i)
    if i.endswith('00'):
        prize_00 += 1
        print(i, end = " ")
    elif i.endswith('0'):
        prize_0 += 1
        print(i, end = " ")
    elif i == '777':
        prize_777 = 1
        print(i, end = " ")
    elif i.endswith('77'):
        prize_77 += 1
        print(i, end = " ")
    elif i.endswith('7'):
        prize_7 += 1
        print(i, end = " ")
    elif i == '111':  
        prize_111 += 1
        print(i, end = " ")
    elif i == '333':
        prize_333 += 1
        print(i, end = " ")
    elif i == '555':
        prize_555 += 1
        print(i, end = " ")   
    elif i == '999':
        prize_999 += 1
        print(i, end = " ") 
    
total_win = prize_0*1 + prize_00*2 + prize_7*3 + prize_77*5 + prize_111*10 + prize_333*15 + prize_555*50 + prize_999*100 + prize_777*200

print('')
print('-'*50)      
print('Общая сумма всех выигрышей', total_win, 'руб.')
print('Кол-во возможных выигрышей по 1 руб.', prize_0)
print('Кол-во возможных выигрышей по 2 руб.', prize_00)
print('Кол-во возможных выигрышей по 3 руб.', prize_7)
print('Кол-во возможных выигрышей по 5 руб.', prize_77)
print('Кол-во возможных выигрышей по 10 руб.', prize_111)
print('Кол-во возможных выигрышей по 15 руб.', prize_333)
print('Кол-во возможных выигрышей по 50 руб.', prize_555)
print('Кол-во возможных выигрышей по 100 руб.', prize_999)
print('Кол-во возможных выигрышей по 200 руб.', prize_777)
print('Количество выигрышных комбинаций', winning_combination)  
print('Средний выигрыш', round(total_win/winning_combination, 2), 'руб.')

000 007 010 017 020 027 030 037 040 047 050 057 060 067 070 077 080 087 090 097 100 107 110 111 117 120 127 130 137 140 147 150 157 160 167 170 177 180 187 190 197 200 207 210 217 220 227 230 237 240 247 250 257 260 267 270 277 280 287 290 297 300 307 310 317 320 327 330 333 337 340 347 350 357 360 367 370 377 380 387 390 397 400 407 410 417 420 427 430 437 440 447 450 457 460 467 470 477 480 487 490 497 500 507 510 517 520 527 530 537 540 547 550 555 557 560 567 570 577 580 587 590 597 600 607 610 617 620 627 630 637 640 647 650 657 660 667 670 677 680 687 690 697 700 707 710 717 720 727 730 737 740 747 750 757 760 767 770 777 780 787 790 797 800 807 810 817 820 827 830 837 840 847 850 857 860 867 870 877 880 887 890 897 900 907 910 917 920 927 930 937 940 947 950 957 960 967 970 977 980 987 990 997 999 
--------------------------------------------------
Общая сумма всех выигрышей 800 руб.
Кол-во возможных выигрышей по 1 руб. 90
Кол-во возможных выигрышей по 2 руб. 10
Кол-во возможных

Таким образом средний выигрыш равен 3,92 рубля. А с учетом вероятности 20% получим:

In [41]:
average_result = 3.92 * 0.2
print(average_result)

0.784


То есть С одного рубля мы теряем 22 копейки.\
ВЫВОД: игра невыгодна

Для проверки решения рассчитаем теоретический средний выигрыш\
X=X1P1+X2P2+...+XnPn

In [43]:
teotetic_average_result = 0.001*(200 + 100 + 50 + 15 + 10 + 5*9 + 3*90 + 2*10 + 1*90)
print(teotetic_average_result)

0.8


Таким образом теоретический и рассчетный результаты практически совпадают.\
0.8 vs 0.784